srcts/src/components/card.ts

```diff
@@ -0,0 +1,277 @@
+      // know exactly where focus is located (we've hidden the button), so we
+      // attach a listener to the next Tab keydown event to entrap focus within
+      // the full screen card.
+      this.lastFocusInterior = this.overlay.anchor;
+      if (!this.container.contains(document.activeElement)) {
+        this.prevFocusExterior = document.activeElement as HTMLElement;
+        this.prevFocusExterior.addEventListener(
+          "keydown",
+          (ev) => this._entrapFocus(ev),
+          { once: true }
+        );
+      }
+    }
+
+    if (this.lastFocusInterior) {
+      this.lastFocusInterior.onkeydown = (ev) => {
+        // If tabbing forwards out of the card, return to close button
+        if (ev.key === "Tab" && !ev.shiftKey) {
+          ev.preventDefault();
+          this.overlay.anchor.focus();
+        }
+      };
+    }
+
+    this.container.classList.add(Card.attr.CLASS_FULL_SCREEN);
+    document.body.classList.add(Card.attr.CLASS_HAS_FULL_SCREEN);
+    this.container.insertAdjacentElement("beforebegin", this.overlay.container);
+  }
+
+  exitFullScreen(): void {
+    // Remove event listeners that were added when entering full screen
+    this.overlay.container.removeEventListener("click", () =>
+      this.exitFullScreen()
+    );
+
+    if (this.lastFocusInterior) {
+      this.lastFocusInterior.onkeydown = null;
+    }
+
+    document.removeEventListener(
+      "keyup",
+      (ev) => this._exitFullScreenOnEscape(ev),
+      false
+    );
+
+    // Remove overlay and remove full screen classes from card
+    this.overlay.container.remove();
+    this.container.classList.remove(Card.attr.CLASS_FULL_SCREEN);
+    document.body.classList.remove(Card.attr.CLASS_HAS_FULL_SCREEN);
+
+    // Reset focus tracking state
+    this.lastFocusInterior = undefined;
+    this.prevFocusExterior = undefined;
+  }
+
+  private _addEventListeners(): void {
+    const btnFullScreen = this.container.querySelector(
+      `:scope > .${Card.attr.CLASS_FULL_SCREEN_ENTER}`
+    );
+    if (!btnFullScreen) return;
+    btnFullScreen.addEventListener("click", (ev) => this.enterFullScreen(ev));
+  }
+
+  private _removeEventListeners(): void {
+    const btnFullScreen = this.container.querySelector(
+      `:scope > .${Card.attr.CLASS_FULL_SCREEN_ENTER}`
+    );
+    if (!btnFullScreen) return;
+    btnFullScreen.removeEventListener("click", (ev) =>
+      this.enterFullScreen(ev)
+    );
+  }
+
+  private _enableTooltips(): void {
+    const selector = `.${Card.attr.CLASS_FULL_SCREEN_ENTER}[data-bs-toggle='tooltip']`;
+    if (!this.container.querySelector(selector)) {
+      return;
+    }
+    const tooltipList = this.container.querySelectorAll(selector);
+    tooltipList.forEach((tt) => new Tooltip(tt));
+  }
+
+  private _exitFullScreenOnEscape(event: KeyboardEvent): void {
+    // if a select input element has focus, then don't exit full screen
+    if (document.activeElement instanceof HTMLSelectElement) {
+      return;
+    }
```

Comment:
I'm not sure this works for selectize.js (if we support this for native select, we just probably also support this for selectize.js?)

Reply:
Thanks for flagging this, let's talk about this behavior in person tomorrow.